_source/_posts/2023-07-28-oidc_workshop.md

```diff
@@ -0,0 +1,505 @@
+---
+layout: blog_post
+title: "Enterprise-Ready Workshop: OpenID Connect"
+author: [edunham, aaron-parecki]
+by: advocate
+communities: [security,javascript]
+description: "Add OpenID Connect support to a sample app, and build the skills to make your production applications more appealing to enterprise customers!"
+tags: [oidc, workshop]
+tweets:
+- ""
+- ""
+- ""
+image: blog/oidc_workshop/social.jpg
+type: awareness
+---
+This workshop is part of our Enterprise Readiness Workshop series. 
+
+In this workshop, you will enhance a sample application to let users access it using their organization's identity provider. When any enterprise customer considers buying your software to enhance their employees' productivity, their IT and security teams want to make sure employees can access your app securely. As a developer, you'd prefer not to rebuild large portions of your authentication flow for every new customer. Fortunately, the OpenID Connect standard solves both of these problems! By adding OpenID Connect (OIDC) support to your app, you can meet the identity security needs of every enterprise organization that uses an OIDC-compatible identity provider. 
+
+Today, we'll walk through adding OIDC to our Todo sample application. 
+
+For an in-depth walkthrough of developing and troubleshooting the code in this workshop, follow along on with the video: 
+{% youtube -gwgEAa4TkU %}
+
+{% include toc.md %}
+
+## Before You Begin: 
+
+Follow the [getting started guide](/blog/2023/07/27/enterprise-ready-getting-started) to set up the Todo app and its dependencies. Make sure you can run the app locally and view its login page in your browser. Launch Prisma studio to examine the database, as well. 
+
+You will also need a free [Okta Developer Account](https://developer.okta.com/login/), to test the setup steps your customer will follow when integrating their OpenID Connect server with your application. 
+
+## The Problem
+
+The Todo app currently only supports password login, but your enterprise customers' IT departments don't want their users managing one-off passwords. You would also like to use customer data, such as which user accounts belong to which organizations, to enhance your app's features. 
+
+Since you value flexibility and maintainability, you want to use a passport library to let your app integrate with arbitrary OpenID Connect servers. 
```

Comment:
```suggestion
Since you value flexibility and maintainability, you want to use a library to let your app integrate with arbitrary OpenID Connect servers. 
```